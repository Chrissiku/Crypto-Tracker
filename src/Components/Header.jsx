import { Link } from 'react-router-dom';
import Icon from 'react-icons-kit';
import { microphone } from 'react-icons-kit/fa/microphone';
import { gear } from 'react-icons-kit/fa/gear';
import logo from '../images/logo.png';

const Header = () => (
  <>
    <header>
      <Link to="/">
        <div className="logo">
          <img src={logo} alt="Crypto Tracker logo" />
          <span>Crypto</span>
        </div>
      </Link>
      <div className="feature">
        <i>
          <Icon size={25} icon={microphone} />
        </i>
        <i>
          <Icon size={25} icon={gear} />
        </i>
      </div>
    </header>
  </>
);

export default Header;
