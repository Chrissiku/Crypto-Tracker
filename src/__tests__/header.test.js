import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Header from '../Components/Header';

it('Test if header render correctly', () => {
  const TREE = TestRenderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>,
  );
  expect(TREE).toMatchSnapshot();
});
