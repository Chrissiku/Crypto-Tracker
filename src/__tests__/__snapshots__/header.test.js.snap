// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if header render correctly 1`] = `
<header>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="logo"
    >
      <img
        alt="Crypto Tracker logo"
        src="logo.png"
      />
      <span>
        Crypto
      </span>
    </div>
  </a>
  <div
    className="feature"
  >
    <i>
      <i
        fill="currentColor"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <svg
          fill="currentColor"
          height={25}
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 1152 1792"
          width={25}
        >
          <path
            d="M1152 704v128q0 221-147.5 384.5t-364.5 187.5v132h256q26 0 45 19t19 45-19 45-45 19h-640q-26 0-45-19t-19-45 19-45 45-19h256v-132q-217-24-364.5-187.5t-147.5-384.5v-128q0-26 19-45t45-19 45 19 19 45v128q0 185 131.5 316.5t316.5 131.5 316.5-131.5 131.5-316.5v-128q0-26 19-45t45-19 45 19 19 45zM896 320v512q0 132-94 226t-226 94-226-94-94-226v-512q0-132 94-226t226-94 226 94 94 226z"
          />
        </svg>
      </i>
    </i>
    <i>
      <i
        fill="currentColor"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <svg
          fill="currentColor"
          height={25}
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 1536 1792"
          width={25}
        >
          <path
            d="M1024 896q0-106-75-181t-181-75-181 75-75 181 75 181 181 75 181-75 75-181zM1536 787v222q0 12-8 23t-20 13l-185 28q-19 54-39 91 35 50 107 138 10 12 10 25t-9 23q-27 37-99 108t-94 71q-12 0-26-9l-138-108q-44 23-91 38-16 136-29 186-7 28-36 28h-222q-14 0-24.5-8.5t-11.5-21.5l-28-184q-49-16-90-37l-141 107q-10 9-25 9-14 0-25-11-126-114-165-168-7-10-7-23 0-12 8-23 15-21 51-66.5t54-70.5q-27-50-41-99l-183-27q-13-2-21-12.5t-8-23.5v-222q0-12 8-23t19-13l186-28q14-46 39-92-40-57-107-138-10-12-10-24 0-10 9-23 26-36 98.5-107.5t94.5-71.5q13 0 26 10l138 107q44-23 91-38 16-136 29-186 7-28 36-28h222q14 0 24.5 8.5t11.5 21.5l28 184q49 16 90 37l142-107q9-9 24-9 13 0 25 10 129 119 165 170 7 8 7 22 0 12-8 23-15 21-51 66.5t-54 70.5q26 50 41 98l183 28q13 2 21 12.5t8 23.5z"
          />
        </svg>
      </i>
    </i>
  </div>
</header>
`;
